scripts/lib/get_tests.py

```diff
@@ -14,124 +14,195 @@
+# Here are the tests belonging to cime suites. Format for individual tests is
+# <test>.<grid>.<compset>[.<testmod>]
+#
+# suite_name : {
+#     "inherit" : (suite1, suite2, ...), # Optional. Suites to inherit tests from. Default is None. Tuple, list, or str.
+#     "time"    : "HH:MM:SS",            # Optional. Recommended upper-limit on test time.
+#     "share"   : True|False,            # Optional. If True, all tests in this suite share a build. Default is False.
+#     "tests"   : (test1, test2, ...)    # Optional. The list of tests for this suite. See above for format. Tuple, list, or str.
+# }
 
 _CIME_TESTS = {
 
-    "cime_tiny" : (None, "0:10:00",
-                   ("ERS.f19_g16_rx1.A",
-                    "NCK.f19_g16_rx1.A")
-                   ),
-
-    "cime_test_only_pass" : (None, "0:10:00",
-                   ("TESTRUNPASS_P1.f19_g16_rx1.A",
-                    "TESTRUNPASS_P1.ne30_g16_rx1.A",
-                    "TESTRUNPASS_P1.f45_g37_rx1.A")
-                   ),
-
-    "cime_test_only_slow_pass" : (None, "0:10:00",
-                   ("TESTRUNSLOWPASS_P1.f19_g16_rx1.A",
-                    "TESTRUNSLOWPASS_P1.ne30_g16_rx1.A",
-                    "TESTRUNSLOWPASS_P1.f45_g37_rx1.A")
-                   ),
-
-    "cime_test_only" : (None, "0:10:00",
-                   ("TESTBUILDFAIL_P1.f19_g16_rx1.A",
-                    "TESTBUILDFAILEXC_P1.f19_g16_rx1.A",
-                    "TESTRUNFAIL_P1.f19_g16_rx1.A",
-                    "TESTRUNSTARCFAIL_P1.f19_g16_rx1.A",
-                    "TESTRUNFAILEXC_P1.f19_g16_rx1.A",
-                    "TESTRUNPASS_P1.f19_g16_rx1.A",
-                    "TESTTESTDIFF_P1.f19_g16_rx1.A",
-                    "TESTMEMLEAKFAIL_P1.f09_g16.X",
-                    "TESTMEMLEAKPASS_P1.f09_g16.X")
-                   ),
-
-    "cime_test_all" : ("cime_test_only", "0:10:00",
-                       ("TESTRUNDIFF_P1.f19_g16_rx1.A", )),
-
-    "cime_developer" : (None, "0:15:00",
-                            ("NCK_Ld3.f45_g37_rx1.A",
-                             "ERI.f09_g16.X",
-                             "ERIO.f09_g16.X",
-                             "SEQ_Ln9.f19_g16_rx1.A",
-                             ("ERS.ne30_g16_rx1.A","drv-y100k"),
-                             "IRT_N2.f19_g16_rx1.A",
-                             "ERR.f45_g37_rx1.A",
-                             "ERP.f45_g37_rx1.A",
-                             "SMS_D_Ln9_Mmpi-serial.f19_g16_rx1.A",
-                             "DAE.ww3a.ADWAV",
-                             "PET_P4.f19_f19.A",
-                             "PEM_P4.f19_f19.A",
-                             "SMS.T42_T42.S",
-                             "PRE.f19_f19.ADESP",
-                             "PRE.f19_f19.ADESP_TEST",
-                             "MCC_P1.f19_g16_rx1.A",
-                             "LDSTA.f45_g37_rx1.A")
-                            ),
+    "cime_tiny" : {
```

Comment:
Changing this format was a significant change but should have no impacts to CESM.